alpha/fsm.cpp

```diff
@@ -0,0 +1,102 @@
+#include "fsm.h"
+#include "hal.h"
+
+_TS_FSM TS_FSM;
+
+// Ctor
+_TS_FSM::_TS_FSM():
+  state_lp([this]() { _TS_FSM::lp_on_enter(); },
+           [this]() { _TS_FSM::lp_on();       },
+           [this]() { _TS_FSM::lp_on_exit();  }),
+            
+  state_hp([this]() { _TS_FSM::hp_on_enter(); },
+           [this]() { _TS_FSM::hp_on();       },
+           [this]() { _TS_FSM::hp_on_exit();  }),
+  fsm(&state_lp)
+{
+  last_print = 0;
+  interval = 1000;
+}
+
+//
+// private fsm state functions
+//
+
+void _TS_FSM::lp_on_enter()
+{ 
+  Serial.println("- Enter LP -"); 
+  current_state = TS_PowerState::LOW_POWER;
+}
+
+void _TS_FSM::lp_on()
+{
+  if (millis() - last_print > interval)
+  {
+    Serial.println("STATE: LOW POWER");
```

Comment:
can we convert all Serial.println to log_i
log_e for errors
log_w for warnings